### DISCUSSION TITLE
Ethernet connection to homelab + bluetooth

### DISCUSSION THREAD

**Discussion by auggie-lahey** [meshtastic/Discussions#108] [General]
I'm very new here, but i recently got into meshtastic for farm automation and disaster relief. I've got one node running and 10 on the way to distibute to family an neighbors for christmas. The initial contribution i want to make is to package the web client for publishing via start9, however there are some requirements that i need help finding/fulfilling. 

Since the nodes will need to communicate with the web client, they would ideally need to be connected via ethernet (to the start9 node) AND bluetooth (phone). I'm unsure if this is possible as the documentation states only bluetooth OR wifi connections can be maintained simultaneously. This is not a deal breaker, as wifi connections to both are possible, but I think the ability to plug the meshtastic node into the ethernet port of the home lab or router, with an antenna outside of the home would be ideal for UX. 

TLDR:
1. is ethernet AND bluetooth connections simultaneously possible? (hardware recommendations appreciated)
2. is it possible to connect meshtastic node to router and then connect to node via web client from another device on that network?
3. if these don't work and i self host the application on my wifi network, then my laptop connects to the node via the self-hosted app, how else can i grab the messages from the node? ie i want to store the messages locally for automation purposes.

**Comment by garthvh** [comment]
The devices support one connection type at a time.

**Reply by auggie-lahey** [reply]
so does this imply i cannot connect to the hardware via my router?

**Reply by garthvh** [reply]
> so does this imply i cannot connect to the hardware via my router?

You can't connect via two methods at once, there is a single supported Ethernet model

**Comment by mariotti** [comment]
Your design is not totally clear to me, but did you consider running your own MQTT server?
I am not sure what UIs are around for MQTT but maybe you can have a look.

**Comment by madeofstown** [comment]
Not sure if you're using Android or Apple but the Android app supports connections to a node over the network (IP). 

Right now the Apple app looks nicer and has a bit better data visualization but the Android app has the better backend that allows more management options.

**Reply by auggie-lahey** [reply]
how do i get the ip?

**Comment by marshalleq** [comment]
This is in my opinion a bit of a problem.  I'm relatively new, but of course straight away I have realised that bluetooth doesn't have the range I need and if I use WiFi there is no phone app meaning messages could be there and I'd never know until I log in.  I don't know why the phone app doesn't just connect over wifi as an option.  The only thing I can think of doing is to have a second device that goes with the phone.
